packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java

```diff
@@ -182,11 +194,22 @@ public void onScanCompleted(String path, Uri uri) {
     this.fileUtils = fileUtils;
   }
 
+  void retrieveLostImage(MethodChannel.Result result) {
+    Map resultMap = resultMapFromPendingResult();
+    if (resultMap.isEmpty()) {
+      result.success(null);
+    } else {
+      result.success(resultMap);
+    }
+    cleanPendingResultPref();
+  }
+
   public void chooseVideoFromGallery(MethodCall methodCall, MethodChannel.Result result) {
     if (!setPendingMethodCallAndResult(methodCall, result)) {
       finishWithAlreadyActiveError(result);
       return;
     }
+    setType("video");
```

Comment:
> Also: is there any way we could unit all the cache logic into a single point and always route through it? Right now it's scattered throughout all these different methods as we're going through the processes depending on the specific image/video type, and then at different points in the process of picking them. It would be hard for anybody adding an extra type here or changing these methods to realize the full flow and realize that they need to extend it or maintain it for whatever extra channel that they're building.

Strongly agree. I am also 'disgusted' by all these scattered code of handling persistent storage right now. I might have to completely change my code structure here or even refactor the old code a bit in order to do so. I will ping you once I have some idea.